lib/src/rdb/datastore.rs

```diff
@@ -418,6 +418,39 @@ impl RocksdbDatastore {
 
         let metadata_manager = MetadataManager::new(&db);
         let indexed_properties = metadata_manager.get_indexed_properties()?;
+        drop(metadata_manager);
+
+        Ok(Database::new(RocksdbDatastore {
+            db: Arc::new(db),
+            indexed_properties: Arc::new(RwLock::new(indexed_properties)),
+        }))
+    }
+
+    /// Creates a new rocksdb datastore with user-tuned rocksdb Option.
+    ///
+    /// # Arguments
+    /// * `path`: The file path to the rocksdb database.
+    /// * `opts`: The user-tuned rocksdb option
+    pub fn new_db_with_options<P: AsRef<Path>>(path: P, opts: Options) -> Result<Database<RocksdbDatastore>> {
```

Comment:
This new function makes sense. With its introduction, though, we should remove `max_open_files` from `new_db`. This is a great time to do it because the main branch contains the unreleased breaking v4 changes anyways.

Reply:
Great. 
Another suggestion: 
What about exposing the `RocksdbDatastore::db` to the lib user? 
-> user can use some rocks helpers to do more low-level magic things, 
e.g. use [`rocksdb::backup::`](https://docs.rs/rocksdb/latest/rocksdb/backup/struct.BackupEngine.html) for backup and [`DBCommon::ingest_external_file_cf`](https://docs.rs/rocksdb/latest/rocksdb/struct.DBCommon.html#method.ingest_external_file_cf) for enhance import work.

Reply:
Not 100% sure if I follow, but for backups and fast imports, those would be useful as extensions, similar to how we have the `repair` fn already.

Reply:
A little different understanding. I consider the lib is for developers who have strong custmizing demond.
They may have more low-level need besides backup and import beyond our expect. Exposing `db` will allow them design their solution before our implement.

Reply:
That seems fairly dangerous to me, since there’s all sorts of ways someone could accidentally corrupt a database by violating assumptions baked into IndraDB. But there’s nothing stopping us from adding this functionality via safe IndraDB APIs.

Reply:
I suppose a power user could also always add this functionality by writing separate tools that use rocksdb directly, but interact with a persisted IndraDB instance.

Reply:
Get your view. It is reasonable.